### PR TITLE
feat(forms): be able to pass custom widgets as props

### PR DESCRIPTION
**Pull request by frassinier**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Not be able to registrer custom widgets from final apps 

**What is the new behavior?**
We can pass custom widget while passing ui specs

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jare-talend** [comment]
:+1:

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -114,6 +114,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   71:21  error  Missing trailing comma                                    comma-dangle
   77:16  error  'formContext' PropType is defined but prop is never used  react/no-unused-prop-types
 
+/home/travis/build/Talend/ui/packages/forms/src/Form.js
+  109:33  error  'widgets' is missing in props validation  react/prop-types
```
Comment (jmfrancois) [inline output/forms.eslint.txt]: please fix this lint error

File: packages/forms/src/Form.js [hunk 2]
```diff
@@ -104,6 +104,11 @@ class Form extends React.Component {
 			throw Error('You must provide data with valid JSON Schema');
 		}
 
+		const widgets = {
+			...customWidgets,
+			...(this.props && this.props.widgets),
```
Comment (jmfrancois) [inline packages/forms/src/Form.js]: question: this.props can really be undefined ?

Comment (frassinier) [inline packages/forms/src/Form.js]: You are right!

File: output/forms.eslint.txt (hunk 1 above)
Comment (frassinier) [inline output/forms.eslint.txt]: Nice spot

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://353.talend.surge.sh)

**Pull request merged**